test/e2e/mirror_test.go

```diff
@@ -0,0 +1,160 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	config "github.com/openshift/api/config/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	batch "k8s.io/api/batch/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
+)
+
+const (
+	testMirrorSetName               = "e2e-powershell-mirror"
+	upstreamPowershellImageLocation = "mcr.microsoft.com/powershell"
+	badPowershellImageLocation      = "test.registry.io/powershell"
+)
+
+func testImageMirroring(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	require.NoError(t, tc.loadExistingNodes(), "error getting the current Windows nodes in the cluster")
+
+	if !t.Run("Mirror settings applied to nodes", tc.testMirrorSettingsApplied) {
+		// No point in running the rest of the tests if settings aren't applied
+		return
+	}
+	t.Run("Workload using a mirrored container image", tc.testWorkloadWithMirroredImage)
```

Comment:
Please do not make this a separate test.
Rather setup the mirrors as part of the test setup, and we should see the tests on a disconnected job pass as normal.

You can make the mirror setup configurable via a CMD line option,

Reply:
Not quite sure what you mean here

- Should we only test mirroring through the disconnected job?
- Do you forsee the mirror_test suite to fail in the disconnected job?

Reply:
> Should we only test mirroring through the disconnected job?

I don't see it necessary to test it elsewhere if the disconnected job is required.

> Do you forsee the mirror_test suite to fail in the disconnected job?

I'm not sure what you mean by this.